community-build/test/scala/dotty/communitybuild/CommunityBuildTest.scala

```diff
@@ -75,7 +75,7 @@ class CommunityBuildTestC:
   // @Test def oslibWatch = projects.oslibWatch.run()
   @Test def playJson = projects.playJson.run()
   @Test def pprint = projects.pprint.run()
-  @Test def protoquill = projects.protoquill.run()
+  // @Test def protoquill = projects.protoquill.run()  // needs to fix new type inference errors
```

Comment:
Disabling protoquill like this is risky: we have at least one open PR which benefits highly from being tested against it (https://github.com/lampepfl/dotty/pull/12540 /cc @nicolasstucki), how much work do you think it would take to fix the new errors?

Reply:
I created a protoquill branch with workarounds for this change:

* https://github.com/dotty-staging/protoquill/commit/1cc61547a4ed77553ec600e28cab78d536786d4d
* https://github.com/dotty-staging/protoquill/tree/workaroud-typing-changes-in-dotty-%2314043

It seems that it is a problem with the `insert` function overloading. I will try to find a better fix.

Reply:
The alternative is to change the API of protoquill to avoid the problematic `insert` overload as I id in https://github.com/getquill/protoquill/commit/5ed25aac9440eb0fcca5d9cd828dc85728fee422.

I suggest using https://github.com/dotty-staging/protoquill/commit/1cc61547a4ed77553ec600e28cab78d536786d4d for this PR as it is less intrusive.

Reply:
Thanks for looking into this, could you propose one of these changes to the author of protoquill?

Reply:
@deusaquilus this PR affects `protoquill`. Could you have a look at the discussion above?

Reply:
Let's go with dotty-staging/protoquill@1cc6154 and encourage @deusaquilus to drop the insert overloading. But since that's an API change it's more complicated.

Reply:
@nicolasstucki Can you make a PR for the protoquill change in the CB (or push a commit as part of this PR) then we can re-enable protoquill here and merge? Since this PR represents a tightening of the typing rules, we should get it into 3.1.2 I hope.

Reply:
So the problem occurs for an untyped `lift` call? Would it work if `lift` was not a macro?

Reply:
If there's no other choice I'll begin the api-change process. Probably need to release a version of Scala2-Quill with the modified API first since I want them to be aligned.

Reply:
Sorry to be a pester and ask again. Would it help the situation at all if `lift` was not a macro but a regular method?

Reply:
I don't know. @nicolasstucki do you have an idea?

Reply:
From what I remember, making `lift` a normal function might help.

Reply:
I also pushed the simple workaround (1cc61547a4ed77553ec600e28cab78d536786d4d) to this PR. We can change the code in protoquill now or after merging this PR. It might be easier to modify protoquill once we have a nightly build with this change.